ssr/src/page/token/info.rs

```diff
@@ -126,47 +143,82 @@ fn TokenInfoInner(
+                    <Show when= move || is_user_principal>
                         <a
                             href=format!("/token/transfer/{root}")
-                            class="flex flex-row justify-self-center justify-center text-white md:text-lg w-full md:w-1/2 rounded-full p-3 bg-primary-600"
+                            class="fixed bottom-20 left-4 right-4 p-3 bg-primary-600 text-white text-center md:text-lg rounded-full z-50"
                         >
                             Send
                         </a>
                     </Show>
-                </AuthCansProvider>
+                {if key_principal.is_some() {
+                    view! { <Transactions source=IndexOrLedger::Index(meta.index) key_principal symbol=meta.symbol.clone()/> }
+                } else {
+                    view! {
+                        <Transactions
+                            source=IndexOrLedger::Ledger(meta.ledger)
+                            key_principal=None
+                            symbol=meta.symbol.clone()
+                        />
+                    }
+                }}
             </div>
         </div>
     }
 }
-
+#[derive(Params, PartialEq, Clone, Serialize, Deserialize)]
+pub struct TokenKeyParam {
+    key_principal: String,
+}
 #[component]
 pub fn TokenInfo() -> impl IntoView {
     let params = use_params::<TokenInfoParams>();
+    let key_principal = use_params::<TokenKeyParam>();
+    let key_principal = move || {
+        key_principal.with(|p| {
+            let TokenKeyParam { key_principal, .. } = p.as_ref().ok()?;
+            Principal::from_text(key_principal).ok()
+        })
+    };
+    let token_metadata_fetch = create_resource(
+        move || (params(), key_principal()),
+        move |(params, key_principal)| async move {
+            let Ok(params) = params else {
+                return Ok::<_, ServerFnError>(None);
+            };
+            // let principal = params.user_principal.to_text().clone();
+            // let root = params.token_root.to_text().clone();
 
-    let token_metadata_fetch = create_resource(params, move |params| async move {
-        let Ok(params) = params else {
-            return Ok::<_, ServerFnError>(None);
-        };
-        // let principal = params.user_principal.to_text().clone();
-        // let root = params.token_root.to_text().clone();
-
-        let cans = unauth_canisters();
-        let meta = token_metadata_by_root(&cans, params.user_principal, params.token_root).await?;
+            let cans = unauth_canisters();
+            let meta = token_metadata_by_root(&cans, key_principal, params.token_root).await?;
 
-        Ok(meta.map(|m| (m, (params.token_root, params.user_principal))))
-    });
+            Ok(meta.map(|m| (m, params.token_root, key_principal)))
+        },
+    );
 
+    let current_user_principal = authenticated_canisters().derive(
```

Comment:
Since the rendering is anyways dependent on this resource, its better to do all the work inside a single resource.
Feel free to use utilities from the futures crate if you're trying to achieve concurrency here

Reply:
So do i combine token_metadata_fetch and current_user_principal into one? or do i directly use the current_user_principal directly into the inner component?

Reply:
combining them makes sense as from past experience more resources = more headaches

If you want them to happen concurrently:
https://docs.rs/futures/latest/futures/future/fn.join.html

Reply:
concurrency wouldnt be required imo because there is only one async task here